### PR TITLE
clang-format: run indent-all

### PR DESCRIPTION
**Pull request by tamiko**

### CODE REVIEW

File: contrib/python-bindings/include/mapping_wrapper.h [hunk 1]
```diff
@@ -67,7 +67,7 @@ namespace python
     PointWrapper
     project_real_point_to_unit_point_on_face(CellAccessorWrapper &cell,
                                              const unsigned int   face_no,
-                                             PointWrapper &       point);
+                                             PointWrapper        &point);
```
Comment (drwells) [inline contrib/python-bindings/include/mapping_wrapper.h:70]: 🎉

File: include/deal.II/lac/petsc_ts.h [hunk 2]
```diff
@@ -317,17 +317,14 @@ namespace PETScWrappers
...
-       Vec>)&&(concepts::is_dealii_petsc_matrix_type<PMatrixType> ||
-               std::constructible_from<
-                 PMatrixType,
-                 Mat>)&&(concepts::is_dealii_petsc_matrix_type<AMatrixType> ||
-                         std::constructible_from<AMatrixType, Mat>))
+    requires((concepts::is_dealii_petsc_vector_type<VectorType> ||
+              std::constructible_from<VectorType, Vec>) &&
+             (concepts::is_dealii_petsc_matrix_type<PMatrixType> ||
+              std::constructible_from<PMatrixType, Mat>) &&
+             (concepts::is_dealii_petsc_matrix_type<AMatrixType> ||
```
Comment (bangerth) [inline include/deal.II/lac/petsc_ts.h:324]: We were talking about this in the other PR: It isn't that the requires stuff is wrapped in a macro, but that the tool (in that version) just doesn't know how to wrap these expressions.

Comment (tamiko) [reply include/deal.II/lac/petsc_ts.h:324]: Yes, I agree that clang-format-16 is much better at handling this. I think the main issue is the fact that the hand-rolled lexer components get massively confused and we see these nonsensical `)&&(` which should just never happen.

It seems important that the clang-format encounters a `requires` statement. If I replace `requires` by, say `#define required requires` then I get the old, wrong behavior:
```
+ 318 #  define required requires                                                       
  319 #  if defined(DEAL_II_HAVE_CXX20) && \                                            
  320     !defined(DEAL_II_DOXYGEN_DO_NOT_PARSE_REQUIRES_CLAUSES)                      
~ 321   required(                                                                       
~ 322     (concepts::is_dealii_petsc_vector_type<VectorType> ||                         
~ 323      std::constructible_from<                                                     
~ 324        VectorType,                                                                
~ 325        Vec>)&&(concepts::is_dealii_petsc_matrix_type<PMatrixType> ||              
~ 326                std::constructible_from<                                           
+ 327                  PMatrixType,                                                     
+ 328                  Mat>)&&(concepts::is_dealii_petsc_matrix_type<AMatrixType> ||    
+ 329                          std::constructible_from<AMatrixType, Mat>))
```

I think what does the trick here is clang-format-16 + `cpp-20` standard set in the clang-format configuration and encountering the `requires` token.

Comment (bangerth) [reply include/deal.II/lac/petsc_ts.h:324]: Hm. What do you think how we can work around this, without having to add hundreds of `#ifdef`s?

**Pull request closed without merging**